### ISSUE TITLE
Cpu selected over GPU when running ollama service

### ISSUE THREAD

**Issue by Talleyrand-34** [ollama/ollama#4614]
### What is the issue?

The problems is that i cannot specify to use gpu over cpu. The instructions to get the gpu running are also not clear. 
Is it that the gpu is not supported? If yes which options do i have?
```
ollama ps
NAME                    ID              SIZE    PROCESSOR       UNTIL
codellama:latest        8fdf8f752f6e    5.1 GB  100% CPU        4 minutes from now
```
systemctl service 
```
[Unit]
Description=Ollama Service
After=network-online.target

[Service]
Enviroment="OLLAMA_MODELS=/home/deck/.ollama/models"
ExecStart=/usr/local/bin/ollama serve
User=ollama
Group=ollama
Restart=always
RestartSec=3
Environment="PATH=/root/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/var/lib/flatpak/exports/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin"
Environment="OLLAMA_HOST=0.0.0.0"
Environment="OLLAMA_DEBUG=1"
#Enviroment="HSA_OVERRIDE_GFX_VERSION=gfx1030"

[Install]
WantedBy=default.target
```
systemctl status
```
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_linux.go:243 msg="amdgpu memory" gpu=0 available="1.0 GiB"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /opt/rocm/lib"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/lib64"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/local/bin/rocm"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/share/ollama/lib/rocm"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=DEBUG source=amd_linux.go:292 msg="rocm supported GPUs" types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:296 msg="amdgpu is not supported" gpu=0 gpu_type=gfx1033 library=/usr/share/ollama/lib/rocm supported_types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:298 msg="See https://github.com/ollama/ollama/blob/main/docs/gpu.md#overrides for HSA_OVERRIDE_GFX_VERSION usage"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=amd_linux.go:311 msg="no compatible amdgpu devices detected"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=types.go:71 msg="inference compute" id=0 library=cpu compute="" driver=0.0 name="" total="14.5 GiB" available="2.8 GiB"
```

GPU info
```
04:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] VanGogh [AMD Custom GPU 0405] (rev ae) (prog-if 00 [VGA controller])
        Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] VanGogh [AMD Custom GPU 0405]
        Flags: bus master, fast devsel, latency 0, IRQ 41
        Memory at f8e0000000 (64-bit, prefetchable) [size=256M]
        Memory at f8f0000000 (64-bit, prefetchable) [size=2M]
        I/O ports at 1000 [size=256]
        Memory at 80300000 (32-bit, non-prefetchable) [size=512K]
        Capabilities: [48] Vendor Specific Information: Len=08 <?>
        Capabilities: [50] Power Management version 3
        Capabilities: [64] Express Legacy Endpoint, MSI 00
        Capabilities: [a0] MSI: Enable- Count=1/4 Maskable- 64bit+
        Capabilities: [c0] MSI-X: Enable+ Count=4 Masked-
        Capabilities: [100] Vendor Specific Information: ID=0001 Rev=1 Len=010 <?>
        Capabilities: [270] Secondary PCI Express
        Capabilities: [2b0] Address Translation Service (ATS)
        Capabilities: [2c0] Page Request Interface (PRI)
        Capabilities: [2d0] Process Address Space ID (PASID)
        Capabilities: [410] Physical Layer 16.0 GT/s <?>
        Capabilities: [440] Lane Margining at the Receiver <?>
        Kernel driver in use: amdgpu
        Kernel modules: amdgpu
```

I am using the steamdeck
### OS

Linux // Arch linux 

### GPU

AMD

### CPU

AMD

### Ollama version

0.1.39

**Comment by falmanna** [comment]
Same here, 

I am running wsl2 and was running fine, I upgraded my Nvidia driver yesterday and suddenly models are only using cpu. I don't see any errors in logs.

Ollama version: 0.1.38
Installation: docker
NVIDIA-SMI 555.42.03
Driver Version: 555.85
CUDA Version: 12.5

**Comment by falmanna** [comment]
UPDATE: Using llama.cpp directly is properly utilizing my gpu.

**Comment by Talleyrand-34** [comment]
> Same here,
> 
> I am running wsl2 and was running fine, I upgraded my Nvidia driver yesterday and suddenly models are only using cpu. I don't see any errors in logs.
> 
> Ollama version: 0.1.38 Installation: docker NVIDIA-SMI 555.42.03 Driver Version: 555.85 CUDA Version: 12.5

I suggest activate debug and paste how ollama is choosing the methos as in here

```
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_linux.go:243 msg="amdgpu memory" gpu=0 available="1.0 GiB"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /opt/rocm/lib"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/lib64"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/local/bin/rocm"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/share/ollama/lib/rocm"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=DEBUG source=amd_linux.go:292 msg="rocm supported GPUs" types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:296 msg="amdgpu is not supported" gpu=0 gpu_type=gfx1033 library=/usr/share/ollama/lib/rocm supported_types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:298 msg="See https://github.com/ollama/ollama/blob/main/docs/gpu.md#overrides for HSA_OVERRIDE_GFX_VERSION usage"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=amd_linux.go:311 msg="no compatible amdgpu devices detected"
May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=types.go:71 msg="inference compute" id=0 library=cpu compute="" driver=0.0 name="" total="14.5 GiB" available="2.8 GiB"
```

**Comment by falmanna** [comment]
Attached
[debug.log](https://github.com/ollama/ollama/files/15448749/debug.log)

**Comment by pdevine** [comment]
@falmanna your issue is unrelated to this one. nVidia updated their drivers w/ version 555 and it's not loading the correct kernel drivers.

This is fixed in #4652 but the workaround for now is to run:

```
sudo modprobe nvidia
sudo modprobe nvidia_uvm
```

@Talleyrand-34 what is the model for your AMD card?

**Comment by Talleyrand-34** [comment]
It is the amd gpu of the steam deck, VanGogh. 
The hardware complete info is the first message @pdevine 

> ```
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_linux.go:243 msg="amdgpu memory" gpu=0 available="1.0 GiB"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /opt/rocm/lib"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.970+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/lib64"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/local/bin/rocm"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.975+02:00 level=DEBUG source=amd_common.go:16 msg="evaluating potential rocm lib dir /usr/share/ollama/lib/rocm"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=DEBUG source=amd_linux.go:292 msg="rocm supported GPUs" types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:296 msg="amdgpu is not supported" gpu=0 gpu_type=gfx1033 library=/usr/share/ollama/lib/rocm supported_types="[gfx1030 gfx1100 gfx1101 gfx1102 gfx900 gfx906 gfx908 gfx90a gfx940 gfx941 gfx942]"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=WARN source=amd_linux.go:298 msg="See https://github.com/ollama/ollama/blob/main/docs/gpu.md#overrides for HSA_OVERRIDE_GFX_VERSION usage"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=amd_linux.go:311 msg="no compatible amdgpu devices detected"
> May 24 20:10:14 steamdeck ollama[43437]: time=2024-05-24T20:10:14.977+02:00 level=INFO source=types.go:71 msg="inference compute" id=0 library=cpu compute="" driver=0.0 name="" total="14.5 GiB" available="2.8 GiB"
> ```
> 
> GPU info
> 
> ```
> 04:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] VanGogh [AMD Custom GPU 0405] (rev ae) (prog-if 00 [VGA controller])
>         Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] VanGogh [AMD Custom GPU 0405]
>         Flags: bus master, fast devsel, latency 0, IRQ 41
>         Memory at f8e0000000 (64-bit, prefetchable) [size=256M]
>         Memory at f8f0000000 (64-bit, prefetchable) [size=2M]
>         I/O ports at 1000 [size=256]
>         Memory at 80300000 (32-bit, non-prefetchable) [size=512K]
>         Capabilities: [48] Vendor Specific Information: Len=08 <?>
>         Capabilities: [50] Power Management version 3
>         Capabilities: [64] Express Legacy Endpoint, MSI 00
>         Capabilities: [a0] MSI: Enable- Count=1/4 Maskable- 64bit+
>         Capabilities: [c0] MSI-X: Enable+ Count=4 Masked-
>         Capabilities: [100] Vendor Specific Information: ID=0001 Rev=1 Len=010 <?>
>         Capabilities: [270] Secondary PCI Express
>         Capabilities: [2b0] Address Translation Service (ATS)
>         Capabilities: [2c0] Page Request Interface (PRI)
>         Capabilities: [2d0] Process Address Space ID (PASID)
>         Capabilities: [410] Physical Layer 16.0 GT/s <?>
>         Capabilities: [440] Lane Margining at the Receiver <?>
>         Kernel driver in use: amdgpu
>         Kernel modules: amdgpu
> ```
>

**Comment by dhiltgen** [comment]
Steam Deck (gfx1033) support is tracked via #3243
